### PR TITLE
add new kurl_util function to get an addon spec as json

### PR DESCRIPTION
**Pull request by laverya**
example output,  from the default 'latestl yaml spec:
```
./kurl_util/bin/yamlutil -j -fp build/default.yaml -jf "spec.contour"
{"version":"latest"}
./kurl_util/bin/yamlutil -j -fp build/default.yaml -jf "spec"
{"contour":{"version":"latest"},"docker":{"version":"latest"},"ekco":{"version":"latest"},"kubernetes":{"version":"latest"},"prometheus":{"version":"latest"},"registry":{"version":"latest"},"rook":{"version":"latest"},"weave":{"version":"latest"}}
```

### CODE REVIEW

File: kurl_util/cmd/yamlutil/main.go [hunk 1]
```diff
@@ -99,20 +101,71 @@ func retrieveField(filePath, yamlPath string) {
...
+	if err != nil {
+		log.Fatalf("error: %v", err)
+	}
+
+	fields := strings.Split(jsonPath, ".")
+
+	// get the specified field
+	for _, field := range fields {
+		concrete := parsed.(map[interface{}]interface{})
+		parsed = concrete[field]
```
Comment (emosbaugh) [inline kurl_util/cmd/yamlutil/main.go:120]: should you do some safety checking here that the field exists with a nice error message?

**Review by emosbaugh — CHANGES_REQUESTED** [review]
Wanna start the good practice of adding tests for these? Seems like an obvious one

File: kurl_util/cmd/yamlutil/main.go (hunk 1 above)
Comment (laverya) [reply kurl_util/cmd/yamlutil/main.go:120]: sure, makes sense

**Review by emosbaugh — APPROVED** [review]
are these tests run in ci

**Pull request merged**